src/code42cli/cmds/search/extraction.py

```diff
@@ -61,12 +69,22 @@ def handle_response(response):
                 events = _get_alert_details(sdk, events)
             except Exception as ex:
                 handlers.handle_error(ex)
+
         handlers.TOTAL_EVENTS += len(events)
-        event = None
-        for event in events:
-            output_logger.info(event)
-        if event:
-            last_event_timestamp = extractor._get_timestamp_from_item(event)
+        if len(events):
+            format_header = (
+                {key: key.capitalize() for key in events[0].keys()}
+                if display_all
+                else {
+                    key: key.capitalize()
+                    for key in events[0].keys()
+                    if type(events[0][key]) == str
+                }
+            )
+
+            click.echo_via_pager(output_format(events, format_header))
+
+            last_event_timestamp = extractor._get_timestamp_from_item(events[-1])
```

Comment:
is this understanding fine? last timestamp is based on the last observed event. Looking at the existing code it seems so, Irrespective of whether it was sorted by some other field, etc, etc.

Reply:
I think this makes sense, yes, but I don't follow what you mean by `sorted by some other field`. I think this has to be sorted by a timestamp to work properly.

Reply:
Yea, that is what I wanted to confirm, extractor always returns result sorted by timestamp, that it doesn't support sorting by other fields.